R/preprocessing.R

```diff
@@ -18,20 +18,34 @@
 #'
 #' @examples
 #' ww_data <- tibble::tibble(
-#'   date = rep(c("2023-11-01", "2023-11-02"), 2),
+#'   date = lubridate::ymd(rep(c("2023-11-01", "2023-11-02"), 2)),
 #'   site = c(rep(1, 2), rep(2, 2)),
 #'   lab = c(1, 1, 1, 1),
 #'   conc = c(345.2, 784.1, 401.5, 681.8),
-#'   lod = c(20, 20, 15, 15)
+#'   lod = c(20, 20, 15, 15),
+#'   site_pop = c(rep(2e5, 2), rep(4e5, 2))
 #' )
-
 #' ww_data_preprocessed <- preprocess_ww_data(ww_data,
-#'                                            conc_col_name = "conc",
-#'                                            lod_col_name = "lod"
-#'                                            )
+#'   conc_col_name = "conc",
+#'   lod_col_name = "lod"
+#' )
 preprocess_ww_data <- function(ww_data,
                                conc_col_name = "genome_copies_per_ml",
                                lod_col_name = "lod") {
+  # This checks that we have all the right column names
+  check_required_ww_inputs(
```

Comment:
This comment suggests the function could be more meaningfully named. Maybe `check_all_needed_ww_columns_present`?

Reply:
Sure can change to this for both colname checks